game/src/game/player/mod.rs

```diff
@@ -50,7 +51,10 @@ impl Plugin for PlayerPlugin {
         app.add_systems(Startup, load_dash_asset);
         app.add_systems(
             GameTickUpdate,
-            ((setup_player, despawn_dead_player)
+            ((
+                setup_player,
+                despawn_dead_player.after(super::game_over::on_game_over),
```

Comment:
It might be cleaner to use a `SystemSet` for ordering this but not really necessary.